File: 1-250/leetcode1.py
# First Method - Use this
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in prevMap:
                return [prevMap[diff],i]
            prevMap[n] = i
        return 

# Second Method
class Solution:
    def twoSum(nums: List[int], target: int) -> List[int]:
        num_set = set()

        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_set:
                complement_index = nums.index(complement)
                return [complement_index, i]
            num_set.add(num)
        return []

File: Leetcode - Go/1-250/leetcode1.go
func twoSum(nums []int, target int) []int {
	prevMap := make(map[int]int)

	for i, n := range nums {
		diff := target - n
		if j, ok := prevMap[diff]; ok {
			return []int{j, i}
		}
		prevMap[n] = i
	}
	return []int{}
}